rust/lance-index/src/scalar.rs

```diff
@@ -139,4 +139,11 @@ pub trait ScalarIndex: Send + Sync + std::fmt::Debug + Index {
     async fn load(store: Arc<dyn IndexStore>) -> Result<Arc<Self>>
     where
         Self: Sized;
+
+    /// Remap the row ids, creating a new remapped version of this index in `dest_store`
+    async fn remap(
```

Comment:
tiny nit: should we move this trait to `Index`?

Reply:
Ideally, yes.  But `ScalarIndex` and `VectorIndex` both handle remapping a little differently today.  `ScalarIndex` interprets it as "remap the files" (thus it is async) and `VectorIndex` interprets it as "remap what's in memory right now" (and thus it is sync).  I think there is some refactoring that needs to happen for these definitions to align.